src/otx/core/model/detection.py

```diff
@@ -163,24 +165,23 @@ def load_state_dict(self, ckpt: dict[str, Any], *args, **kwargs) -> None:
 
     def _log_metrics(self, meter: Metric, key: Literal["val", "test"], **compute_kwargs) -> None:
         if key == "val":
-            retval = super()._log_metrics(meter, key)
-
-            # NOTE: Validation metric logging can update `best_confidence_threshold`
-            if best_confidence_threshold := getattr(meter, "best_confidence_threshold", None):
-                self.hparams["best_confidence_threshold"] = best_confidence_threshold
+            # NOTE: Validation metric logging can update `best_confidence_threshold`,
+            # if there's no pre-defined best confidence threshold
+            if self.confidence_threshold is None:
+                retval = super()._log_metrics(meter, key)
+                if best_confidence_threshold := getattr(meter, "best_confidence_threshold", None):
+                    self.hparams["best_confidence_threshold"] = best_confidence_threshold
+            else:
+                retval = super()._log_metrics(meter, key, best_confidence_threshold=self.confidence_threshold)
+                self.hparams["best_confidence_threshold"] = self.confidence_threshold
```

Comment:
According to the naming change,
```suggestion
            # if there's no fixed confidence threshold
            if fixed_confidence_threshold := self.hparam.get("fixed_confidence_threshold", None):
                retval = super()._log_metrics(meter, key, best_confidence_threshold=fixed_confidence_threshold)
                self.hparams["best_confidence_threshold"] = fixed_confidence_threshold
            else:
                retval = super()._log_metrics(meter, key)
                self.hparams["best_confidence_threshold"] = getattr(meter, "best_confidence_threshold", None)
```

Reply:
This logic will save best_confidence_threshold and fixed_confidence_threshold filed, and the value of both filed would be same if users specify their own threshold. Is it right?

Reply:
Hmm, I thought that you wanted to make them same. If not they are equal to each other, we cannot do the test with the fixed confidence threshold because your test logic is
https://github.com/openvinotoolkit/training_extensions/blob/01a07f6ecd16db153f5c4d6cd81717f253325ab9/src/otx/core/model/detection.py#L180-L184

Under this implementation, we have to set `best_confidence_threshold = fixed_confidence_threshold`.

Yea, personally, I prefer the two values ​​to be independent each other. If so, we need to change this part too. `fixed_confidence_threshold` should be get prioritized than `best_confidence_threshold` when it exists during the test phase.

Reply:
How about fix hparam's field name as confidence threshold, if user specify fixed confidence threshold then confidence threshold would be fixed confidence threshold if not, it will be best confidence threshold from metric. And test phase will  only take confidence threshold from hparam or xml file